user.go

```diff
@@ -13,28 +15,41 @@ import (
 // UserController implements the user resource.
 type UserController struct {
 	*goa.Controller
-	identityRepository account.IdentityRepository
-	tokenManager       token.Manager
+	db           application.DB
+	tokenManager token.Manager
 }
 
 // NewUserController creates a user controller.
-func NewUserController(service *goa.Service, identityRepository account.IdentityRepository, tokenManager token.Manager) *UserController {
-	return &UserController{Controller: service.NewController("UserController"), identityRepository: identityRepository, tokenManager: tokenManager}
+func NewUserController(service *goa.Service, db application.DB, tokenManager token.Manager) *UserController {
+	return &UserController{Controller: service.NewController("UserController"), db: db, tokenManager: tokenManager}
 }
 
 // Show returns the authorized user based on the provided Token
 func (c *UserController) Show(ctx *app.ShowUserContext) error {
-	identID, err := c.tokenManager.Locate(ctx)
+	id, err := c.tokenManager.Locate(ctx)
 	if err != nil {
 		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(err.Error()))
 		return ctx.BadRequest(jerrors)
 	}
-	ident, err := c.identityRepository.Load(ctx, identID)
-	if err != nil {
-		fmt.Printf("Auth token contains id %s of unknown Identity\n", identID)
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(fmt.Sprintf("Auth token contains id %s of unknown Identity\n", identID)))
-		return ctx.Unauthorized(jerrors)
-	}
 
-	return ctx.OK(ident.ConvertIdentityFromModel())
+	return application.Transactional(c.db, func(appl application.Application) error {
+		identity, err := appl.Identities().Load(ctx, id)
+		if err != nil || identity == nil {
+			log.Printf("Auth token contains id %s of unknown Identity\n", id)
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(fmt.Sprintf("Auth token contains id %s of unknown Identity\n", id)))
+			return ctx.Unauthorized(jerrors)
```

Comment:
you can replace the 2 lines above with:
````
return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorized(fmt.Sprintf("Auth token contains id %s of unknown Identity\n", id), err))
````

But you'll need to follow the existing code in `error/errors.go` to implement the `errors.NewUnauthorized(...)` function.